Profiler/Basics/ProfilerEntry.cs

```diff
@@ -0,0 +1,70 @@
+using System.Threading;
+using Profiler.Core;
+using Profiler.Util;
+
+namespace Profiler.Basics
+{
+    /// <summary>
+    /// Paired with BaseProfiler, provide a summary of computation time per a key object. 
+    /// </summary>
+    public sealed class ProfilerEntry
+    {
+        long _totalMainThreadTimeMs;
+        long _totalOffThreadTimeMs;
+
+        // Use Pool
+        ProfilerEntry()
+        {
+        }
+
+        /// <summary>
+        /// Total main-thread computation time of the game associated with the key object in milliseconds.
+        /// </summary>
+        public long TotalMainThreadTimeMs => _totalMainThreadTimeMs;
+
+        /// <summary>
+        /// Total not-main-thread computation time of the game associated with the key object in milliseconds.
+        /// </summary>
+        public long TotalOffThreadTimeMs => _totalOffThreadTimeMs;
+
+        /// <summary>
+        /// Total computation time of the game associated with the key object in milliseconds.
+        /// </summary>
+        public long TotalTimeMs => TotalMainThreadTimeMs + TotalOffThreadTimeMs;
+
+        internal void Add(ProfilerResult profilerResult)
+        {
+            if (profilerResult.IsMainThread)
```

Comment:
If you're only adding to a profiler entry from a single thread (which you would with a central producer/consumer queue of profiler results) then you don't need interlocked.